### ISSUE TITLE
[Question] nomad docker task use consul dns server

### ISSUE THREAD

**Issue by zd1990** [hashicorp/nomad#7279]
hi, i have a question for the docker task,how can i use consul dns server in docker task,the config like this can not work
```
    task "filebeat" {           # The task stanza specifices a task (unit of work) within a group  
      driver = "docker"      # This task uses Docker, other examples: exec, LXC, QEMU  
      env {
        LOGPATH="/alloc/logs/test_v10.0.0_task*"
        APPID="test"
        TAGS="v10.0.0"
        TOPIC="log_nomad"
      }
      config {
        image = "filebeat_nomad:v6"  # Docker image to download (uses public hub by default)
        dns_servers = ["127.0.0.1"]
      }
      resources {      # Resource limits to reserve for this task
        cpu    = 100   # 500 MHz, capacity is calculated by (CPU Freq * Num CPUs)
        memory = 100   # 256MB
      }
    }
```
error:
```
2020/03/05 15:03:41.006939 log.go:12: WARN Failed to connect to broker kafka.service.consul:9092: dial tcp: lookup kafka.service.consul on 127.0.0.1:53: read udp 127.0.0.1:48905->127.0.0.1:53: read: connection refused
2020/03/05 15:03:41.006956 log.go:16: WARN kafka message: client/metadata got error from broker while fetching metadata:%!(EXTRA *net.OpError=dial tcp: lookup kafka.service.consul on 127.0.0.1:53: read udp 127.0.0.1:48905->127.0.0.1:53: read: connection refused)
2020/03/05 15:03:41.006963 log.go:16: WARN kafka message: client/metadata no available broker to send metadata request to
```

**Comment by zd1990** [comment]
i should set the network mode to host,thks
```
network_mode = "host"
```

**Comment by chulkilee** [comment]
If you came here from search...

- `127.0.0.1` on host is not exposed to containers as `127.0.0.1`
- Option 1: dnsmasq or coredns or whatever forwarding dns server on private ip address of host, and pass the private ip address to containers as dns server
- Option 2: make consul dns works on host with systemd-resolved (which runs 127.0.0.53:53) and then use user-defined docker network, which uses the host dns server. (Note that 127.0.0.53 is not visible to containers)

**Comment by PinkLolicorn** [comment]
If you came here from a search like I did...

Just go to the official Consul docs on this topic: https://learn.hashicorp.com/tutorials/consul/dns-forwarding

😎
